InterGalacticCurrencyConvertor/src/main/java/materials/MaterialFactory.java

```diff
@@ -0,0 +1,55 @@
+package main.java.materials;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Keeps track of different types of materials that can be sold InterGalactically.
+ */
+public final class MaterialFactory {
+  private HashMap<String, Material> mMtlMap  = new HashMap<>();
+  private static MaterialFactory mInstance = new MaterialFactory();
+  public static MaterialFactory getInstance() {
+    return mInstance;
+  }
+  public void add(String name, int numUnits, double totalCost) {
+    Objects.requireNonNull(name, "Invalid Material Name");
+    mMtlMap.put(name, new Material(name, numUnits, totalCost));
+  }
+  public boolean isAvailable(String name) {
+    if (name != null) {
+      name = name.toLowerCase();
+    }
+    return mMtlMap.containsKey(name);
+  }
+  public double getUnitPrice(String name) {
+    Objects.requireNonNull(name, "Invalid Material Name");
+    if (!isAvailable(name)) {
+      throw new IllegalArgumentException("Material does not exist");
```

Comment:
Include name in the exception - since it will help in debugging.

Reply:
done